### PR TITLE
Add a clarification for the public_baseurl property.

### PR DESCRIPTION
**Pull request by zblesk**
I have been a bit confused about the correct way to set the `public_baseurl` and did it wrong. I've realized the mistake with some help from the folks in #synapse:matrix.org and they suggested I try and add a clarification to the sample file. So that's what I'm trying to do. 

Signed-off-by: Ladislav Benc <ladislav.benc@protonmail.com>

### CODE REVIEW

**Comment by clokep** [comment]
@zblesk Can you make the corresponding change in `synapse/config/server.py` to make CI happy? Thanks!

File: docs/sample_config.yaml [hunk 1]
```diff
@@ -47,7 +47,9 @@ pid_file: DATADIR/homeserver.pid
 # (not including _matrix/...). This is the same URL a user would
 # enter into the 'custom HS URL' field on their client. If you
 # use synapse with a reverse proxy, this should be the URL to reach
-# synapse via the proxy.
+# synapse via the proxy. If you use delegation and your synapse URL
```
Comment (clokep) [inline docs/sample_config.yaml:50]:
```suggestion
# synapse via the proxy. If you use delegation and your synapse URL
```

**Comment by zblesk** [comment]
Well. I tried updating until it stopped complainng about mismatches and trailing spaces; now it says `No new newsfragments found on this branch.` and I have no idea what _that_ means, sorry.

**Comment by clokep** [comment]
@zblesk Please take a look at the contributing guide: https://github.com/matrix-org/synapse/blob/develop/CONTRIBUTING.md#changelog

**Comment by zblesk** [comment]
As mentioned above, I'm only doing this because gus in the #synapse channel asked me to, after I got stuck on this point. 
If it doesn't fit, feel free to close this PR.

**Comment by clokep** [comment]
Thanks for your contribution. I'm going to close this per @richvdh's comment above.

**Pull request closed without merging**